src/System.Private.CoreLib/shared/System/Globalization/DateTimeParse.cs

```diff
@@ -4621,6 +4636,201 @@ internal static bool TryParseQuoteString(ReadOnlySpan<char> format, int pos, Str
+                    case 0x6f637420 /* 'oct ' */ : month = 10; break;
+                    case 0x6e6f7620 /* 'nov ' */ : month = 11; break;
+                    case 0x64656320 /* 'dec ' */ : month = 12; break;
+                    default:
+                        result.SetBadDateTimeFailure();
+                        return false;
+                }
+            }
+
+            // Parse the four-digit year.
+            int year;
+            {
+                uint y1 = (uint)(source[12] - '0'), y2 = (uint)(source[13] - '0'), y3 = (uint)(source[14] - '0'), y4 = (uint)(source[15] - '0');
+
+                if (y1 > 9 || y2 > 9 || y3 > 9 || y4 > 9)
+                {
+                    result.SetBadDateTimeFailure();
+                    return false;
+                }
+
+                year = (int)(y1*1000 + y2*100 + y3*10 + y4);
+            }
+
+            if (source[16] != ' ')
+            {
+                result.SetBadDateTimeFailure();
+                return false;
+            }
+
+            // Parse the two digit hour.
+            int hour;
+            {
+                uint h1 = (uint)(source[17] - '0'), h2 = (uint)(source[18] - '0');
+
+                if (h1 > 9 || h2 > 9)
+                {
+                    result.SetBadDateTimeFailure();
+                    return false;
+                }
+
+                hour = (int)(h1*10 + h2);
+            }
+
+            if (source[19] != ':')
+            {
+                result.SetBadDateTimeFailure();
+                return false;
+            }
+
+            // Parse the two-digit minute.
+            int minute;
+            {
+                uint m1 = (uint)(source[20] - '0');
+                uint m2 = (uint)(source[21] - '0');
+
+                if (m1 > 9 || m2 > 9)
+                {
+                    result.SetBadDateTimeFailure();
+                    return false;
+                }
+
+                minute = (int)(m1*10 + m2);
+            }
+
+            if (source[22] != ':')
+            {
+                result.SetBadDateTimeFailure();
+                return false;
+            }
+
+            // Parse the two-digit second.
+            int second;
+            {
+                uint s1 = (uint)(source[23] - '0'), s2 = (uint)(source[24] - '0');
+
+                if (s1 > 9 || s2 > 9)
+                {
+                    result.SetBadDateTimeFailure();
+                    return false;
+                }
+
+                second = (int)(s1*10 + s2);
+            }
+
+            // Parse " GMT".  It must be upper case.
+            if (source[25] != ' ' || source[26] != 'G' || source[27] != 'M' || source[28] != 'T')
+            {
+                result.SetBadDateTimeFailure();
+                return false;
+            }
+
+            // Validate that the parsed date is valid according to the calendar.
+            if (!parseInfo.calendar.TryToDateTime(year, month, day, hour, minute, second, 0, 0, out result.parsedDate))
```

Comment:
>if (!parseInfo.calendar.TryToDateTim [](start = 12, length = 36)

This is ok, but with the "r" format we always use Gregorian calendar so we can just simply do "new DateTime(...) " we just need to ensure DateTimeKind value is set correctly.

Reply:
> note that calendar.TryToDateTime is catching the exception and returns false at that time

GregorianCalendar overrides TryToDateTime, with a bit of logic that ends up calling DateTime.TryCreate.  Seems reasonable to still just use calendar.TryToDateTime then and inherit that behavior.  We could avoid a virtual call by rewriting that logic here, but it doesn't really seem worth it.